### ISSUE TITLE
Log an exception if the app is terminated by the OS (Android) to free up memory.

### ISSUE THREAD

**Issue by oodavid** [flutter/flutter#93917]
## Use case

TLDR; logging errors if the app is terminated by Android to free up memory would help us fix errors in the wild.

There has been a long-standing issue with `image_picker` that has been hard to diagnose. @stuartmorgan has shed some light on this issue; Android will sometimes terminate the Flutter app to free up memory. This can occur when a separate intent/activity is in the foreground. There are no errors logged when this occurs.

https://github.com/flutter/flutter/issues/48016#issuecomment-971918009

## Proposal

Throw an error if the app is destroyed to clear up memory.

## Notes

I know very little about Android activities and processes, these shed a little light on where and when they are destroyed for memory.

https://developer.android.com/guide/components/activities/activity-lifecycle#asem
https://developer.android.com/guide/components/activities/process-lifecycle

NB: I have logged a similar request with the crashlytics team in case there's something similar needed for them.

https://github.com/firebase/firebase-android-sdk/issues/3150

I have logged a similar request with the crashlytics team

https://github.com/firebase/firebase-android-sdk/issues/3150

**Comment by stuartmorgan** [comment]
> Throw an error if the app is destroyed to clear up memory.

There is no reliable way to do this.

@blasten Do you happen to know how useful `onDestroy` would be for this purpose? The lifecycle diagrams suggest that it would never be called in this situation (just as on iOS), while the `onDestroy` docs are much more vague ("This can happen [...] because the system is temporarily destroying this instance of the activity to save space." but then "There are situations where the system will simply kill the activity's hosting process without calling this method (or any others) in it"). So it's unclear to me if best-effort logging would ever actually fire in practice.

**Comment by maheshmnj** [comment]
Keeping this issue open for further insights from the team.

**Comment by chinmaygarde** [comment]
I agree with Stuart that there isn't a reliable way to do this. Closing as cant-fix.

**Comment by dnfield** [comment]
In particular

> There are situations where the system will simply kill the activity's hosting process without calling this method (or any others) in it, so it should not be used to do things that are intended to remain around after the process goes away.

https://developer.android.com/reference/android/app/Activity#onDestroy()

**Comment by dnfield** [comment]
And we don't know in onDestroy why we're getting destroyed...
